cedar-policy-core/src/authorizer/err.rs

```diff
@@ -32,3 +32,12 @@ pub enum AuthorizationError {
         error: EvaluationError,
     },
 }
+
+impl AuthorizationError {
+    /// Get the id of the erroring policy
+    pub fn id(&self) -> &PolicyID {
```

Comment:
Should this not  be `pub` since `AuthorizationError` is `pub use`'ed but this return the internal `PolicyId` type?

The enum already exposes this to anyone who matches on it, but returning it from a function could be more confusing IMO. Better to just have the function on `Diagnostics` that returns the wrapped `PolicyId` type.

Reply:
Responding to your earlier comment: 

> Rather than adding an error_ids method to Diagnostics, it would make more sense to add an id method to the publicly exposed AuthorizationError type that returns a PolicyId. I don’t think this is possible without a breaking change.

Agreed. I'm not sure how we would structure it, but the ideal situation would be a public `AuthorizationError` type with a function returning the public `PolicyId` type, but I also don't think that could be done without a breaking change. For now I'd say we drop this function.

Reply:
Good catch. No, we don't want to expose a function that returns a `PolicyID`. I'll make a separate PR to change the `AuthorizationError` definition in the next major release.